test/cmd/images.sh

```diff
@@ -20,125 +21,125 @@ USE_IMAGES=${USE_IMAGES:-$defaultimage}
-[ -n "$(oc get imageStreams wildfly --template="{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}")" ]
-oc get istag | grep -q "wildfly"
-oc annotate istag/wildfly:latest foo=bar
-oc get istag/wildfly:latest -o jsonpath={.metadata.annotations.foo} | grep -q "bar"
-oc annotate istag/wildfly:latest foo-
-[ ! "$(oc get istag/wildfly:latest -o jsonpath={.metadata.annotations} | grep -q 'bar')" ]
-oc delete imageStreams ruby
-oc delete imageStreams nodejs
-oc delete imageStreams wildfly
-oc delete imageStreams postgresql
-oc delete imageStreams mongodb
-[ -z "$(oc get imageStreams ruby --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams nodejs --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams postgresql --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams mongodb --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams wildfly --template="{{.status.dockerImageRepository}}")" ]
-tryuntil oc get imagestreamTags mysql:latest
-[ -n "$(oc get imagestreams mysql --template="{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}")" ]
-oc describe istag/mysql:latest
-[ "$(oc describe istag/mysql:latest | grep "Environment:")" ]
-[ "$(oc describe istag/mysql:latest | grep "Image Created:")" ]
-[ "$(oc describe istag/mysql:latest | grep "Image Name:")" ]
+os::cmd::expect_success "tryuntil 'oc get imagestreamtags wildfly:latest'"
+os::cmd::expect_success_and_text "oc get imageStreams wildfly --template='{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}'" '[0-9]{4}\-[0-9]{2}\-[0-9]{2}' # expect a date like YYYY-MM-DD
+os::cmd::expect_success_and_text 'oc get istag' 'wildfly'
+os::cmd::expect_success 'oc annotate istag/wildfly:latest foo=bar'
+os::cmd::expect_success_and_text 'oc get istag/wildfly:latest -o jsonpath={.metadata.annotations.foo}' 'bar'
+os::cmd::expect_success 'oc annotate istag/wildfly:latest foo-'
+os::cmd::expect_success_and_not_text 'oc get istag/wildfly:latest -o jsonpath={.metadata.annotations}' 'bar'
+os::cmd::expect_success 'oc delete imageStreams ruby'
+os::cmd::expect_success 'oc delete imageStreams nodejs'
+os::cmd::expect_success 'oc delete imageStreams wildfly'
+os::cmd::expect_success 'oc delete imageStreams postgresql'
+os::cmd::expect_success 'oc delete imageStreams mongodb'
+os::cmd::expect_failure 'oc get imageStreams ruby'
+os::cmd::expect_failure 'oc get imageStreams nodejs'
+os::cmd::expect_failure 'oc get imageStreams postgresql'
+os::cmd::expect_failure 'oc get imageStreams mongodb'
+os::cmd::expect_failure 'oc get imageStreams wildfly'
+os::cmd::expect_success "tryuntil 'oc get imagestreamTags mysql:latest'"
+os::cmd::expect_success_and_text "oc get imagestreams mysql --template='{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}'" '[0-9]{4}\-[0-9]{2}\-[0-9]{2}' # expect a date like YYYY-MM-DD
```

Comment:
This problem does not happen a couple of lines earlier with wildfly image, because that one contains only one tag, for now at least.